### PR TITLE
Replace deprecated unittest aliases with supported functions

### PR DESCRIPTION
**Pull request by rodrigc**
See: https://docs.python.org/2.7/library/unittest.html#deprecated-aliases

https://twistedmatrix.com/trac/ticket/8371

### CODE REVIEW

**Comment by rodrigc** [comment]
This command `2to3 -f asserts` was used to identify the places that needed to be fixed.

File: twisted/test/test_application.py [hunk 1]
```diff
@@ -85,29 +85,29 @@ def testDisowning(self):
 
     def testRunning(self):
         s = service.Service()
-        self.assert_(not s.running)
+        self.assertTrue(not s.running)
```
Comment (adiroiban) [inline twisted/test/test_application.py]: maybe self.assertFalse :)

File: twisted/test/test_application.py [hunk 2]
```diff
@@ -212,7 +212,7 @@ def testProcessComponent(self):
 
     def testServiceComponent(self):
         a = service.Application("hello")
-        self.assert_(service.IService(a) is service.IServiceCollection(a))
+        self.assertTrue(service.IService(a) is service.IServiceCollection(a))
```
Comment (adiroiban) [inline twisted/test/test_application.py]: maybe self.assertIs

File: twisted/test/test_defer.py [hunk 3]
```diff
@@ -511,12 +511,12 @@ def testCallbackErrors(self):
         l = []
         d = defer.Deferred().addCallback(lambda _: 1 // 0).addErrback(l.append)
         d.callback(1)
-        self.assert_(isinstance(l[0].value, ZeroDivisionError))
+        self.assertTrue(isinstance(l[0].value, ZeroDivisionError))
```
Comment (adiroiban) [inline twisted/test/test_defer.py]: maybe self.assertIsInstance

File: twisted/test/test_loopback.py [hunk 4]
```diff
@@ -103,8 +103,8 @@ def makeConnection(self, transport):
         server = TestProtocol()
         client = TestProtocol()
         loopback.loopbackAsync(server, client)
-        self.failIfEqual(client.transport, None)
-        self.failIfEqual(server.transport, None)
+        self.assertNotEqual(client.transport, None)
```
Comment (adiroiban) [inline twisted/test/test_loopback.py]: maybe self.assertIsNotNone

File: twisted/test/test_compat.py [hunk 5]
```diff
@@ -651,7 +651,7 @@ def test_reraiseWithNone(self):
         except:
             typ2, value2, tb2 = sys.exc_info()
             self.assertEqual(typ2, ZeroDivisionError)
-            self.assertTrue(value is value2)
+            self.assertTrue(value, value2)
```
Comment (adiroiban) [inline twisted/test/test_compat.py]: maybe asertIs ?

File: twisted/test/test_failure.py [hunk 6]
```diff
@@ -540,10 +540,10 @@ def test_cleanFailureRemovesTracebackInPython3(self):
         the exception to C{None} in Python 3.
         """
         f = getDivisionFailure()
-        self.assertNotEqual(f.tb, None)
-        self.assertIdentical(f.value.__traceback__, f.tb)
+        self.assertIsNotNone(f.tb, None)
```
Comment (adiroiban) [inline twisted/test/test_failure.py]: just self.assertIsNotNone(f.tb)

File: twisted/test/test_compat.py (hunk 5 above)
Comment (rodrigc) [inline twisted/test/test_compat.py]: Yes.

File: twisted/test/test_failure.py (hunk 6 above)
Comment (rodrigc) [inline twisted/test/test_failure.py]: Yes, see latest commit.

File: twisted/words/test/test_irc.py [hunk 7]
```diff
@@ -1417,7 +1417,7 @@ def removeFeature(name):
             msg = "are available on this server"
             self._serverTestImpl(
                 '005', msg, 'isupport', args=name, options=[name])
-            self.assertIdentical(
+            self.assertIs(
```
Comment (adiroiban) [inline twisted/words/test/test_irc.py]: maybe assertIsNone ?

Comment (rodrigc) [inline twisted/words/test/test_irc.py]: OK.  See latest commit.

**Comment by rodrigc** [comment]
Based on discussion with @adiroiban on IRC, this patch is a bit too big
to digest for review purposes. :)
I will submit as multiple smaller patches.  Which can be reviewed and incorporated separately.

**Comment by rodrigc** [comment]
I split this patch into multiple patches to make it easier to review:
- https://github.com/twisted/twisted/pull/92
- https://github.com/twisted/twisted/pull/93
- https://github.com/twisted/twisted/pull/94
- https://github.com/twisted/twisted/pull/95
- https://github.com/twisted/twisted/pull/96
- https://github.com/twisted/twisted/pull/97
- https://github.com/twisted/twisted/pull/98
- https://github.com/twisted/twisted/pull/99
- https://github.com/twisted/twisted/pull/100
- https://github.com/twisted/twisted/pull/101
- https://github.com/twisted/twisted/pull/102
- https://github.com/twisted/twisted/pull/103
- https://github.com/twisted/twisted/pull/104
- https://github.com/twisted/twisted/pull/105
- https://github.com/twisted/twisted/pull/106
- https://github.com/twisted/twisted/pull/107
- https://github.com/twisted/twisted/pull/108
- https://github.com/twisted/twisted/pull/109
- https://github.com/twisted/twisted/pull/110

**Pull request closed without merging**